include/circt-c/Dialect/OM.h

```diff
@@ -96,42 +83,54 @@ MLIR_CAPI_EXPORTED MlirModule omEvaluatorGetModule(OMEvaluator evaluator);
 //===----------------------------------------------------------------------===//
 
 /// Query if the Object is null.
-MLIR_CAPI_EXPORTED bool omEvaluatorObjectIsNull(OMObject object);
+MLIR_CAPI_EXPORTED bool omEvaluatorObjectIsNull(OMEvaluatorValue object);
 
 /// Get the Type from an Object, which will be a ClassType.
-MLIR_CAPI_EXPORTED MlirType omEvaluatorObjectGetType(OMObject object);
+MLIR_CAPI_EXPORTED MlirType omEvaluatorObjectGetType(OMEvaluatorValue object);
 
 /// Get a field from an Object, which must contain a field of that name.
-MLIR_CAPI_EXPORTED OMObjectValue omEvaluatorObjectGetField(OMObject object,
-                                                           MlirAttribute name);
+MLIR_CAPI_EXPORTED OMEvaluatorValue
+omEvaluatorObjectGetField(OMEvaluatorValue object, MlirAttribute name);
 
 /// Get all the field names from an Object, can be empty if object has no
 /// fields.
 MLIR_CAPI_EXPORTED MlirAttribute
-omEvaluatorObjectGetFieldNames(OMObject object);
+omEvaluatorObjectGetFieldNames(OMEvaluatorValue object);
 
 //===----------------------------------------------------------------------===//
-// ObjectValue API.
+// EvaluatorValue API.
 //===----------------------------------------------------------------------===//
 
-// Query if the ObjectValue is null.
-MLIR_CAPI_EXPORTED bool omEvaluatorObjectValueIsNull(OMObjectValue objectValue);
+// Query if the EvaluatorValue is null.
+MLIR_CAPI_EXPORTED bool omEvaluatorValueIsNull(OMEvaluatorValue evaluatorValue);
 
-/// Query if the ObjectValue is an Object.
+/// Query if the EvaluatorValue is an Object.
 MLIR_CAPI_EXPORTED bool
-omEvaluatorObjectValueIsAObject(OMObjectValue objectValue);
-
-/// Get the Object from an  ObjectValue, which must contain an Object.
-MLIR_CAPI_EXPORTED OMObject
-omEvaluatorObjectValueGetObject(OMObjectValue objectValue);
+omEvaluatorValueIsAObject(OMEvaluatorValue evaluatorValue);
 
-/// Query if the ObjectValue is a Primitive.
+/// Query if the EvaluatorValue is a Primitive.
 MLIR_CAPI_EXPORTED bool
-omEvaluatorObjectValueIsAPrimitive(OMObjectValue objectValue);
+omEvaluatorValueIsAPrimitive(OMEvaluatorValue evaluatorValue);
 
-/// Get the Primitive from an  ObjectValue, which must contain a Primitive.
+/// Get the Primitive from an EvaluatorValue, which must contain a Primitive.
 MLIR_CAPI_EXPORTED MlirAttribute
-omEvaluatorObjectValueGetPrimitive(OMObjectValue objectValue);
+omEvaluatorValueGetPrimitive(OMEvaluatorValue evaluatorValue);
+
+/// Get the EvaluatorValue from a Primitive value.
+MLIR_CAPI_EXPORTED OMEvaluatorValue
+omEvaluatorValueFromPrimitive(MlirAttribute primitive);
+
+/// Query if the EvaluatorValue is an Object.
+MLIR_CAPI_EXPORTED bool
+omEvaluatorValueIsAList(OMEvaluatorValue evaluatorValue);
+
+/// Get the length of the list.
+MLIR_CAPI_EXPORTED intptr_t
+omListGetNumElements(OMEvaluatorValue evaluatorValue);
```

Comment:
nit: I tried to prefix all the C APIs for the evaluator with `omEvaluator`, just in case we ever add C APIs for OM dialect constructs separately from the evaluator.